### PR TITLE
feat(avoidance): change lateral margin based on if it's parked vehicle

### PR DESCRIPTION
**Pull request by satoshi-ota**
## Description

See https://github.com/autowarefoundation/autoware.universe/pull/6520

## Tests performed

- [x] PASS TIER IV INTERNAL SCENARIOS

## Effects on system behavior

Improve avoidance maneuver.

## Pre-review checklist for the PR author

The PR author **must** check the checkboxes below when creating the PR.

- [x] I've confirmed the [contribution guidelines].
- [x] The PR follows the [pull request guidelines].

## In-review checklist for the PR reviewers

The PR reviewers **must** check the checkboxes below before approval.

- [ ] The PR follows the [pull request guidelines].

## Post-review checklist for the PR author

The PR author **must** check the checkboxes below before merging.

- [ ] There are no open discussions or they are tracked via tickets.

After all checkboxes are checked, anyone who has write access can merge the PR.

[contribution guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/
[pull request guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/pull-request-guidelines/

### CODE REVIEW

File: autoware_launch/config/planning/scenario_planning/lane_driving/behavior_planning/behavior_path_planner/avoidance_by_lane_change/avoidance_by_lane_change.param.yaml [hunk 1]
```diff
@@ -12,64 +12,80 @@
...
         bicycle:
           execute_num: 2
           moving_speed_threshold: 0.28                 # 1.0km/h
           moving_time_threshold: 1.0
           max_expand_ratio: 0.0
           envelope_buffer_margin: 0.8
-          avoid_margin_lateral: 0.0
-          safety_buffer_lateral: 1.0
+          lateral_margin:
+            soft_margin: 0.0                           # [m]
```
Comment (TomohitoAndo) [inline autoware_launch/config/planning/scenario_planning/lane_driving/behavior_planning/behavior_path_planner/avoidance_by_lane_change/avoidance_by_lane_change.param.yaml:66]: Would it be acceptable to change this value to 0.0 though the original parameter (`safety_buffer_lateral`) is 1.0 ?

Comment (TomohitoAndo) [reply autoware_launch/config/planning/scenario_planning/lane_driving/behavior_planning/behavior_path_planner/avoidance_by_lane_change/avoidance_by_lane_change.param.yaml:66]: (I mean `safety_buffer_lateral: 1.0` -> `hard_margin: 0.0`)

Comment (satoshi-ota) [reply autoware_launch/config/planning/scenario_planning/lane_driving/behavior_planning/behavior_path_planner/avoidance_by_lane_change/avoidance_by_lane_change.param.yaml:66]: @TomohitoAndo Ah. Thanks for your pointing out. This is a mistake. I'll fix it.

Comment (satoshi-ota) [reply autoware_launch/config/planning/scenario_planning/lane_driving/behavior_planning/behavior_path_planner/avoidance_by_lane_change/avoidance_by_lane_change.param.yaml:66]: I fixed in https://github.com/autowarefoundation/autoware_launch/pull/894/commits/5e4a0e34b093feaa1ba0a1390ca731389aa5c02c.

**Review by TomohitoAndo — APPROVED** [review]
LGTM

**Pull request merged**